lib/tapioca/compilers/dsl/active_record_fixtures.rb

```diff
@@ -0,0 +1,82 @@
+# typed: strict
+# frozen_string_literal: true
+
+begin
+  require "rails"
+  require "active_record"
+  require "active_record/fixtures"
+  require "active_support/test_case"
+rescue LoadError
+  return
+end
+
+module Tapioca
+  module Compilers
+    module Dsl
+      # `Tapioca::Compilers::Dsl::ActiveRecordFixtures` decorates RBIs for test fixture methods
+      # that are created dynamically by Rails.
+      #
+      # For example, given an application with a posts table, we can have a fixture file
+      #
+      # ~~~yaml
+      # first_post:
+      #   author: John
+      #   title: My post
+      # ~~~
+      #
+      # Rails will allow us to invoke `posts(:first_post)` in tests to get the fixture record.
+      # The generated RBI by this generator will produce the following
+      #
+      # ~~~rbi
+      # # test_case.rbi
+      # # typed: true
+      # class ActiveSupport::TestCase
+      #   sig { params(fixture_names: Symbol).returns(Post) }
+      #   def posts(*fixture_names); end
+      # end
+      # ~~~
+      class ActiveRecordFixtures < Base
+        extend T::Sig
+
+        class FixtureLoader
+          include ActiveRecord::TestFixtures
+        end
+
+        sig { override.params(root: RBI::Tree, constant: T.class_of(ActiveSupport::TestCase)).void }
+        def decorate(root, constant)
+          FixtureLoader.fixture_path = Rails.root.join("test", "fixtures")
+          FixtureLoader.fixtures(:all)
+
+          cache = ActiveRecord::FixtureSet::ClassCache.new({}, ActiveRecord::Base)
+          table_names = FixtureLoader.fixture_table_names
+          return if table_names.empty?
+
+          root.create_path(constant) do |mod|
+            table_names.each do |table_name|
+              accessor_name = table_name.tr("/", "_")
+              model_class = cache[table_name]
+
+              create_fixture_method(mod, accessor_name, model_class.name)
```

Comment:
I don't think we can do `model_class.name` as the return type here since the return type is dependent on the number of parameters supplied to the fixture method:
```ruby
posts(:first_post, :last_post) => [#<Post:0x111>, #<Post:0x222>]
```
so this will have to be `T.untyped`, I am afraid.

Reply:
Unfortunately we can't, since the two types are not compatible. Imagine if I called `.map` on the return value, the model would not have that method, so Sorbet will complain about the model type not having the method.